src/NServiceBus.Core/Impersonation/Windows/WindowsIdentityEnricher.cs

```diff
@@ -19,11 +18,7 @@ public void MutateOutgoing(LogicalMessage logicalMessage, TransportMessage trans
                 return;
             }
             var windowsIdentity = WindowsIdentity.GetCurrent();
-            if (windowsIdentity != null)
```

Comment:
All good @danielmarbach, yeah I got confused. You are right `Thread.CurrentPrincipal` was the problematic one.